### PR TITLE
ssh-audit: init at 1.7.0

### PR DESCRIPTION
**Pull request by 4z3**
###### Motivation for this change

[ssh-audit](https://github.com/arthepsy/ssh-audit) is not packaged, yet.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by Lassulus — APPROVED** [review]

**Comment by Lassulus** [comment]
tested with nix-review and checked some configs via `ssh-audit $hostname`

**Review by tomberek — APPROVED** [review]
Builds, runs, functions on Ubuntu. No reverse dependencies.

File: pkgs/tools/security/ssh-audit/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
+{ fetchFromGitHub, python3Packages, stdenv }:
+
+python3Packages.buildPythonPackage rec {
```
Comment (aanderse) [inline pkgs/tools/security/ssh-audit/default.nix]: From the looks this is simply an application and not a library, so you should use `buildPythonApplication` instead.

File: pkgs/tools/security/ssh-audit/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      packages=['ssh-audit'],
+      scripts=['scripts/ssh-audit'],
+      url='${meta.homepage}',
+      version='${version}',
+    )
+  '';
+
+  meta = {
+    description = "tool for ssh server auditing";
+    homepage = "https://github.com/arthepsy/ssh-audit";
```
Comment (aanderse) [inline pkgs/tools/security/ssh-audit/default.nix]: Quoting not required

File: pkgs/tools/security/ssh-audit/default.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    license = stdenv.lib.licenses.mit;
+    maintainers = [
+      stdenv.lib.maintainers.tv
+    ];
+    version = "1.7.0";
```
Comment (aanderse) [inline pkgs/tools/security/ssh-audit/default.nix]: The `version` attribute belongs in the derivation itself, not `meta`.

File: pkgs/tools/security/ssh-audit/default.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      name='${pname}',
+      packages=['ssh-audit'],
+      scripts=['scripts/ssh-audit'],
+      url='${meta.homepage}',
+      version='${version}',
+    )
+  '';
+
+  meta = {
+    description = "tool for ssh server auditing";
```
Comment (aanderse) [inline pkgs/tools/security/ssh-audit/default.nix]: > Don’t include a period at the end. Don’t include newline characters. Capitalise the first character. For brevity, don’t repeat the name of package — just describe what it does.

Please capitalise the first character.

**Review by aanderse — CHANGES_REQUESTED** [review]
Looking good, just a few minor nitpicks.

File: pkgs/tools/security/ssh-audit/default.nix (hunk 2 above)
Comment (Lassulus) [reply pkgs/tools/security/ssh-audit/default.nix]: https://github.com/NixOS/rfcs/pull/45

File: pkgs/tools/security/ssh-audit/default.nix (hunk 1 above)
Comment (Lassulus) [reply pkgs/tools/security/ssh-audit/default.nix]: exports also a library: `import ssh-audit`

**Comment by Lassulus** [comment]
@GrahamcOfBorg build ssh-audit

File: pkgs/tools/security/ssh-audit/default.nix (hunk 1 above)
Comment (aanderse) [reply pkgs/tools/security/ssh-audit/default.nix]: Didn't see that, thanks!

File: pkgs/tools/security/ssh-audit/default.nix (hunk 2 above)
Comment (aanderse) [reply pkgs/tools/security/ssh-audit/default.nix]: I had no idea about that rfc! Thank you for pointing that out to me!

**Pull request merged**